src/mlpack/tests/main_tests/preprocess_split_test.cpp

```diff
@@ -357,3 +357,24 @@ TEST_CASE_METHOD(
   REQUIRE(static_cast<uvec>(find(
       params.Get<arma::Mat<size_t>>("test_labels") == 2)).n_rows == 20);
 }
+
+/**
+ * Check that if input dataset is not empty.
+ * Empty dataset can not be split.
+ */
+TEST_CASE_METHOD(
+    PreprocessSplitTestFixture, "PreprocessSplitNonEmptyInputTest",
+    "[PreprocessSplitMainTest][BindingTests]")
+{
+  // Load custom dataset.
+  arma::mat inputData(0,0);
+  arma::Mat<size_t> labels(0,0);
+
+  SetInputParam("input", std::move(inputData));
+  SetInputParam("input_labels", std::move(labels));
+
+  // Now check that the input dataset is not empty.
+  Log::Fatal.ignoreInput = true;
+  REQUIRE_THROWS_AS(RUN_BINDING() , std::bad_exception);
+  Log::Fatal.ignoreInput = false;
```

Comment:
Looked at the test failure and it's what I would expect, with `Log::Fatal.ignoreInput = true;`, https://github.com/mlpack/mlpack/pull/3140/files#diff-672c29065a6b712d8425fadf512687e92061b406166bcfa62dd5350fc6d90656R337 will not exit but only print the failure, so we continue and call https://github.com/mlpack/mlpack/blob/master/src/mlpack/core/data/split_data.hpp#L157 which fails because the matrix is empty. But in your case you actually want to throw an exception in `RequireNonEmptyInputValue`, because it makes no sense to continue even if we don't use `Log::Fatal`. I'm trying to figure out why we like to continue for some checks, perhaps @rcurtin can provide some input here; possible that I missed something.

Reply:
I think yes that's all about what I have understood so far. As per the [issue reported](https://github.com/mlpack/mlpack/issues/3123), a check for input sizes with a better error message would prevent anybody to use empty arrays or empty datasets because empty datasets can't be split.

Reply:
Agreed, so I think we should throw an exception at that point, and don't continue even if `Fatal` does not exit, I'm still curious why we don't do that for any of the other `Require...` methods.

Reply:
yes, agreed! I am not sure about other ``Require...`` methods.

Reply:
Require should catch that exception, but why is not handling it properly I can't understand.